src/include/duckdb/function/table/arrow.hpp

```diff
@@ -59,7 +63,7 @@ struct ArrowTableFunction {
 	//! Actual conversion from Arrow to DuckDB
 	static void ArrowToDuckDB(ArrowScanState &scan_state,
 	                          std::unordered_map<idx_t, vector<std::pair<ArrowListType, idx_t>>> &arrow_lists,
-	                          DataChunk &output);
+	                          std::unordered_map<idx_t, LogicalType> &dictionary_type, DataChunk &output);
```

Comment:
Rather than adding another unordered_map, perhaps this could be abstracted over, e.g. having something like `unordered_map<idx_t, unique_ptr<ArrowConvertData>>` where `ArrowConvertData` could either be a `ListArrowConvertData` that holds the list type of the column, or a `DictionaryArrowConvertData` that holds the dictionary type.